Mathlib/Algebra/Module/PID.lean

```diff
@@ -269,4 +269,22 @@ theorem equiv_free_prod_directSum [h' : Module.Finite R M] :
           (h.prodCongr g).trans <| LinearEquiv.prodComm.{u, u} R _ (Fin n →₀ R) ⟩⟩
   rw [range_subtype, ker_mkQ]
 
+open LinearMap in
+theorem exists_ker_toSpanSingleton_eq_annihilator [Module.Finite R M] :
+    ∃ x : M, ker (toSpanSingleton R _ x) = annihilator R M := by
```

Comment:
I have a somewhat crazy idea of doing "FG/PID induction", where you could reduce this to proving the theorem about the "standardly presented modules", along with the fact that this property is "isomorphism-invariant"...

Reply:
No, I want the property to be satisfied by all FG/PID modules iff it is satisfied by a module in the standard form